lib/widgets/upload_button.dart

```diff
@@ -0,0 +1,125 @@
+import 'dart:typed_data';
+
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:file_picker/file_picker.dart';
+import 'package:flutter/material.dart';
+import 'package:positioned_tap_detector/positioned_tap_detector.dart';
+
+class UploadButtonController {
+  UploadButtonController(this.defaultImage, {this.onUpdate});
+
+  final ImageProvider defaultImage;
+
+  _UploadButtonState _uploadButtonState;
+  ImageProvider newImg;
```

Comment:
The design of this component is quite odd. It's just an upload button, so why would it ever have to know how to display an image?

Keep in mind the single-responsibility principle: https://en.wikipedia.org/wiki/Single-responsibility_principle

The upload button should know how to build itself, and should be able to provide image data (aka the image bytes, Uint8List). It should never control how other widgets use that data (i.e. how the image is displayed).

Please remove all the ImageProviders and related code from UploadButton & its controller. Each page that uses the UploadButton should choose what it wants to do with the data it returns, and how it wants to display it.